src/test/e2e/psp.feature.ts

```diff
@@ -0,0 +1,185 @@
+      airnodeProtocolAbi.abi,
+      airnodeProtocolAbi.bytecode,
+      roles.deployer
+    );
+    airnodeProtocol = await airnodeProtocolFactory.deploy();
+
+    dapiServerAbi = JSON.parse(fs.readFileSync(path.resolve('./scripts/artifacts/DapiServer.json')).toString());
+    dapiServerFactory = new ethers.ContractFactory(dapiServerAbi.abi, dapiServerAbi.bytecode, roles.deployer);
+    dapiServer = await dapiServerFactory.deploy(
+      accessControlRegistry.address,
+      dapiServerAdminRoleDescription,
+      roles.manager.address,
+      airnodeProtocol.address
+    );
+
+    // Access control
+    const managerRootRole = await accessControlRegistry.deriveRootRole(roles.manager.address);
+    await accessControlRegistry
+      .connect(roles.manager)
+      .initializeRoleAndGrantToSender(managerRootRole, dapiServerAdminRoleDescription);
+
+    // Wallets
+    const airnodeWallet = ethers.Wallet.fromMnemonic(localConfig.airnodeMnemonic);
+    const airnodePspSponsorWallet = node.evm
+      .deriveSponsorWalletFromMnemonic(localConfig.airnodeMnemonic, roles.sponsor.address, PROTOCOL_ID_PSP)
+      .connect(provider);
+    await roles.deployer.sendTransaction({
+      to: airnodePspSponsorWallet.address,
+      value: ethers.utils.parseEther('1'),
+    });
+
+    // Templates
+    const endpointId = ethers.utils.keccak256(
+      ethers.utils.defaultAbiCoder.encode(
+        ['string', 'string'],
+        [localConfig.endpoint.oisTitle, localConfig.endpoint.endpointName]
+      )
+    );
+    const parameters = abi.encode(localConfig.templateParameters);
+    const templateId = ethers.utils.solidityKeccak256(['bytes32', 'bytes'], [endpointId, parameters]);
+
+    // Subscriptions
+    const threshold = (await dapiServer.HUNDRED_PERCENT()).div(localConfig.threshold); // Update threshold %
+    const beaconUpdateSubscriptionConditionParameters = ethers.utils.defaultAbiCoder.encode(['uint256'], [threshold]);
+    const beaconUpdateSubscriptionConditions = [
+      {
+        type: 'bytes32',
+        name: '_conditionFunctionId',
+        value: ethers.utils.defaultAbiCoder.encode(
+          ['bytes4'],
+          [dapiServer.interface.getSighash('conditionPspBeaconUpdate')]
+        ),
+      },
+      { type: 'bytes', name: '_conditionParameters', value: beaconUpdateSubscriptionConditionParameters },
+    ];
+    const encodedBeaconUpdateSubscriptionConditions = abi.encode(beaconUpdateSubscriptionConditions);
+    await dapiServer
+      .connect(roles.randomPerson)
+      .registerBeaconUpdateSubscription(
+        airnodeWallet.address,
+        templateId,
+        encodedBeaconUpdateSubscriptionConditions,
+        airnodeWallet.address,
+        roles.sponsor.address
+      );
+  });
+
+  it('updates the beacon successfully', async () => {
+    jest.spyOn(config, 'loadAirnodeConfig').mockImplementation(() => airnodeConfig as any);
+    jest.spyOn(config, 'loadAirkeeperConfig').mockImplementation(() => airkeeperConfig as any);
```

Comment:
I wasn't able to make these config mocks work with `mockReadFileSync` as the **airnodeConfig** loading keeps failing in process-subscriptions.